[CI] CloneSnapshotIT.testMasterFailoverDuringCloneStep1 failure

**Build scan**:
https://gradle-enterprise.elastic.co/s/a6lovmsnlvypu

**Repro line**:
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.snapshots.CloneSnapshotIT.testMasterFailoverDuringCloneStep1" \
  -Dtests.seed=79DAA05140C620AC \
  -Dtests.security.manager=true \
  -Dtests.locale=sq \
  -Dtests.timezone=Africa/Lagos \
  -Druntime.java=11

**Reproduces locally?**:
no

**Applicable branches**:
master

**Failure history**:
Two more since Oct 5th:
https://gradle-enterprise.elastic.co/s/lpt6ip6vz7sbc
https://gradle-enterprise.elastic.co/s/tpnuofyxi2ucq

**Failure excerpt**:
java.lang.AssertionError: Found leaked snapshot completion listeners {test-repo:target-snapshot/XrAiUKcgSkaoFGtCvH3XMg=[org.elasticsearch.action.ActionListener$1@64a0aff0]} on [{node_t2}{AR7VHs2TRf2LNBefAhu9vA}{xuFfSHmZTF-v9duV1ushWg}{127.0.0.1}{127.0.0.1:46509}{m}]
	at org.elasticsearch.snapshots.SnapshotsService.assertAllListenersResolved(SnapshotsService.java:2239)
	at org.elasticsearch.snapshots.AbstractSnapshotIntegTestCase.lambda$verifyNoLeakedListeners$0(AbstractSnapshotIntegTestCase.java:123)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:937)

--- a/server/src/internalClusterTest/java/org/elasticsearch/snapshots/CloneSnapshotIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/snapshots/CloneSnapshotIT.java
@@ -360,8 +360,9 @@ public class CloneSnapshotIT extends AbstractSnapshotIntegTestCase {
         createFullSnapshot(repoName, sourceSnapshot);
 
         blockMasterOnReadIndexMeta(repoName);
+        final String cloneName = "target-snapshot";
         final ActionFuture<AcknowledgedResponse> cloneFuture =
-                startCloneFromDataNode(repoName, sourceSnapshot, "target-snapshot", testIndex);
+                startCloneFromDataNode(repoName, sourceSnapshot, cloneName, testIndex);
         awaitNumberOfSnapshotsInProgress(1);
         final String masterNode = internalCluster().getMasterName();
         waitForBlock(masterNode, repoName, TimeValue.timeValueSeconds(30L));
@@ -377,6 +378,9 @@ public class CloneSnapshotIT extends AbstractSnapshotIntegTestCase {
 
         awaitNoMoreRunningOperations(internalCluster().getMasterName());
 
+        // Check if the clone operation worked out by chance as a result of the clone request being retried because of the master failover
+        cloneSucceeded = cloneSucceeded ||
+            getRepositoryData(repoName).getSnapshotIds().stream().anyMatch(snapshotId -> snapshotId.getName().equals(cloneName));
         assertAllSnapshotsSuccessful(getRepositoryData(repoName), cloneSucceeded ? 2 : 1);
     }
 

--- a/server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java
+++ b/server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java
@@ -223,6 +223,8 @@ public class SnapshotsService extends AbstractLifecycleComponent implements Clus
         final String repositoryName = request.repository();
         final String snapshotName = indexNameExpressionResolver.resolveDateMathExpression(request.snapshot());
         validate(repositoryName, snapshotName);
+        // TODO: create snapshot UUID in CreateSnapshotRequest and make this operation idempotent to cleanly deal with transport layer
+        //       retries
         final SnapshotId snapshotId = new SnapshotId(snapshotName, UUIDs.randomBase64UUID()); // new UUID for the snapshot
         Repository repository = repositoriesService.repository(request.repository());
         if (repository.isReadOnly()) {
@@ -333,6 +335,8 @@ public class SnapshotsService extends AbstractLifecycleComponent implements Clus
         }
         final String snapshotName = indexNameExpressionResolver.resolveDateMathExpression(request.target());
         validate(repositoryName, snapshotName);
+        // TODO: create snapshot UUID in CloneSnapshotRequest and make this operation idempotent to cleanly deal with transport layer
+        //       retries
         final SnapshotId snapshotId = new SnapshotId(snapshotName, UUIDs.randomBase64UUID());
         final Snapshot snapshot = new Snapshot(repositoryName, snapshotId);
         initializingClones.add(snapshot);

